src/main/java/org/opensearch/security/filter/SecurityRestFilter.java

```diff
@@ -109,28 +117,26 @@ public SecurityRestFilter(
-     * If the request is authenticated, then it goes through a whitelisting check.
-     * The whitelisting check works as follows:
-     * If whitelisting is not enabled, then requests are handled normally.
-     * If whitelisting is enabled, then SuperAdmin is allowed access to all APIs, regardless of what is currently whitelisted.
-     * If whitelisting is enabled, then Non-SuperAdmin is allowed to access only those APIs that are whitelisted in {@link #requests}
-     * For example: if whitelisting is enabled and requests = ["/_cat/nodes"], then SuperAdmin can access all APIs, but non SuperAdmin
+     * If the request is authenticated, then it goes through a allowlisting check.
+     * The allowlisting check works as follows:
+     * If allowlisting is not enabled, then requests are handled normally.
+     * If allowlisting is enabled, then SuperAdmin is allowed access to all APIs, regardless of what is currently allowlisted.
+     * If allowlisting is enabled, then Non-SuperAdmin is allowed to access only those APIs that are allowlisted in {@link #requests}
+     * For example: if allowlisting is enabled and requests = ["/_cat/nodes"], then SuperAdmin can access all APIs, but non SuperAdmin
      * can only access "/_cat/nodes"
-     * Further note: Some APIs are only accessible by SuperAdmin, regardless of whitelisting. For example: /_opendistro/_security/api/whitelist is only accessible by SuperAdmin.
+     * Further note: Some APIs are only accessible by SuperAdmin, regardless of allowlisting. For example: /_opendistro/_security/api/whitelist is only accessible by SuperAdmin.
      * See {@link AllowlistApiAction} for the implementation of this API.
      * SuperAdmin is identified by credentials, which can be passed in the curl request.
      */
     public RestHandler wrap(RestHandler original, AdminDNs adminDNs) {
-        return new RestHandler() {
-
-            @Override
-            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
-                org.apache.logging.log4j.ThreadContext.clearAll();
-                if (!checkAndAuthenticateRequest(request, channel, client)) {
-                    User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-                    if (userIsSuperAdmin(user, adminDNs)
-                        || (whitelistingSettings.checkRequestIsAllowed(request, channel, client)
-                            && allowlistingSettings.checkRequestIsAllowed(request, channel, client))) {
+        return (request, channel, client) -> {
+            org.apache.logging.log4j.ThreadContext.clearAll();
+            if (!checkAndAuthenticateRequest(request, channel)) {
+                User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+                if (userIsSuperAdmin(user, adminDNs)
+                    || (whitelistingSettings.checkRequestIsAllowed(request, channel, client)
```

Comment:
Swap to allowlisting?

Reply:
we are checking for both to be backwards compatible. Check out line below where I've added a check for allowlistSettings